test/port-finder-0-vs-127-test.js

```diff
@@ -0,0 +1,47 @@
+/*
+ * port-finder-0-vs-127-test.js: Test for the `portfinder` module.
+ * that demonstrates issue #24 
+ * https://github.com/indexzero/node-portfinder/issues/24
+ */
+
+var vows = require('vows'),
+    assert = require('assert'),
+    portfinder = require('../lib/portfinder'),
+    child_process = require('child_process'),
+    path = require('path'),
+    http = require('http');
+
+const host = "localhost";
+var server;
+
+vows.describe('portfinder').addBatch({
+  "When using portfinder module": {
+    "ensure that infinite loop is not triggered": {
+      topic: function () {
+    var callback = this.callback;
+    portfinder.getPort(function(err, port) {
+      server = http.createServer(function(){}).listen(port, host, function() {
+        var timeout = false;
+        var fileToExec = path.join(__dirname, 'getPort.js');
+        var timer = setTimeout(function() {
+          timeout = true;
+          process.kill(child.pid);
+          callback(null, "timeout");  
+        }, 10000); // 10 seconds
+        var child = child_process.spawn('node', [fileToExec,  port]);
+        child.on('close', function() {
+          if (timeout === false) {
+            clearTimeout(timer);
+            callback(null);
+          }
+        });
+      });  
+    });
+      },
+      "should not get a timeout": function (err, res) {
+        server.close();
+        assert.isFalse(!!res);
+      }
+    }
+  }
+}).export(module);
```

Comment:
Amending this comment(from previous inline comment on tests:)

> we can add localhost to this array: https://github.com/indexzero/node-portfinder/blob/master/test/helper.js#L27 - i'd actually like to add an exact duplicate ip address there (you can add localhost as well, it should not matter, if our expected behavior works out right).

instead, what i should have said is that we can create a test that looks like this:

```js
    "with 5 existing servers": {
      topic: function () {
        testHelper(servers, this.callback);
      },
      "the getPort({host: '127.0.0.1'}) method": {
        topic: function () {
          portfinder.getPort({ host: '127.0.0.1}, this.callback);
        },
        "should respond with the first free port (32773)": function (err, port) {
          assert.isTrue(!err);
          assert.equal(port, 32773);
        }
      }
    }
```

for say 127.0.0.1 and localhost

such a test falls falls nicely in line with [this current test](https://github.com/indexzero/node-portfinder/blob/master/test/port-finder-test.js#L23-L31), but test a specific host, something we we're not previously doing (thus our lack of coverage lies there) (technically we should perhaps have a 3rd test that tests a host and a port, and that the port starts iterating from the passed port, but we can make an issue for that if your short on time, no biggie)